pandas/core/arrays/datetimelike.py

```diff
@@ -2369,6 +2370,27 @@ def interpolate(
             return self
         return type(self)._simple_new(out_data, dtype=self.dtype)
 
+    def take(
+        self,
+        indices: TakeIndexer,
+        *,
+        allow_fill: bool = False,
+        fill_value: Any = None,
+        axis: AxisInt = 0,
+    ) -> Self:
+        result = super().take(
+            indices=indices, allow_fill=allow_fill, fill_value=fill_value, axis=axis
+        )
+
+        indices = np.asarray(indices, dtype=np.intp)
+        maybe_slice = lib.maybe_indices_to_slice(indices, len(self))
+
+        if isinstance(maybe_slice, slice):
+            freq = self._get_getitem_freq(maybe_slice)
+            result.freq = freq
```

Comment:
i think if you set result._freq it will be more performant

Reply:
changed

Reply:
Hi @jbrockmendel is this now corresponding to your idea?

Reply:
Hi @jbrockmendel , any news on this? would be nice to finish the PR at some point :)